via/services/transcript.py

```diff
@@ -0,0 +1,48 @@
+from io import StringIO
+
+from webvtt import WebVTT
+from webvtt.parsers import SRTParser
+
+from via.services.http import HTTPService
+
+
+class TranscriptService:
+    """Fetch transcripts in WebVTT or SRT formats."""
+
+    def __init__(self, http_service: HTTPService):
+        self._http_service = http_service
+
+    def get_transcript(self, url: str):
+        """Fetch a transcript and return it in JSON format."""
+
+        transcript = self._get_vtt(url)
+        segments = []
+
+        for caption in transcript:
+            start = caption.start_in_seconds
+            end = caption.end_in_seconds
+            segments.append(
+                {"start": start, "duration": end - start, "text": caption.text}
+            )
+
+        return segments
+
+    def _get_vtt(self, url: str) -> WebVTT:
+        response = self._http_service.get(url)
+        content = response.text.strip()
+        content_buf = StringIO(content)
```

Comment:
A malicious user could potentially disrupt the service by passing the URL of a very large transcript file. We could mitigate this by capping the size of the transcript that we're willing to process (eg. take only the first 1 MB).

Reply:
The `response` object here is a `Response` instance from the `requests` library. This can indeed read the response in a streaming fashion.